app/controllers/api/v1/requests_controller.rb

```diff
@@ -49,6 +49,15 @@ def create
     end
   end
 
+  def update
+    plant_attributes_from_params(@request)
+    if @request.update(request_params)
```

Comment:
En mettant les instructions dans cet ordre, on peut s'exposer à un petit souci de cohérence.
La première méthode appelée, `plant_attributes_from_params`, regarde si un ID est passé en paramètre, et assigne les champs `plant_stage_name` et `plant_name` si c'est le cas.

Puis, la méthode `update` assigne tous les paramètres passés par l'utilisateur ; dont les fameux `plant_stage_name` et `plant_name`.

Ce qui veut dire que dans l'idée, la méthode `plant_attributes_from_params` peut ici être overridée par les paramètres passés par le front, et donc créer des valeurs "illogiques" (par exemple, si passe l'id d'une rose mais je mets en dur que le nom est une tulipe ; et ça sera enregistré tel quel dans la bdd).

En inversant l'ordre d'appel des méthodes, on évite cette potentielle "faille" :) 
```suggestion
    @request.assign_attributes(request_params)
    plant_attributes_from_params(@request)

    if @request.save
```
(si tu regardes bien, c'est d'ailleurs bien dans cet ordre que sont appelées les méthodes dans le `create`, ou qu'elles l'étaient dans la méthode `update` d'origine, qui a récemment été supprimée dans une autre PR)